### PR TITLE
Adding support for select region

### PR DESCRIPTION
**Pull request by phpcentre**
Hey there,

This is Bartholomew, this request will add support for users to select any region they want while creating a bucket, the same way you create a bucket on the site s3 console.
If you notice any issue please report back, I  properly checked the functionality, and it's working fine. 

Thanks.

### CODE REVIEW

File: classes/amazon-s3-and-cloudfront.php [hunk 1]
```diff
@@ -1234,6 +1264,16 @@ function translate_region( $region ) {
 		return $region;
 	}
 
+	function select_bucket_location( $location_key = false ) {
+		$locations = array( 'us-east-1' => 'US Standard','us-west-2' => 'Oregon','us-west-1' => 'Northern California','eu-west-1' => 'Ireland','eu-central-1' => 'Frankfurt','ap-southeast-1' => 'Singapore','ap-northeast-1' => 'Tokyo','ap-southeast-2' => 'Sydney','sa-east-1' => 'Sao Paulo' );
```
Comment (bradt) [inline classes/amazon-s3-and-cloudfront.php]: Is there an AWS API request that would return a list of locations instead of hard coding them here?

Comment (phpcentre) [inline classes/amazon-s3-and-cloudfront.php]: I have checked the AWS doc and I cannot find any API that would return a list of the locations. This is what I could find though,the available location constraint for all their services are listed here  http://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region

**Comment by A5hleyRich** [comment]
@phpcentre Thanks for the PR, this will be included in the next release.

**Comment by polevaultweb** [comment]
@phpcentre and thanks for all your help on the support forum!

**Pull request closed without merging**